kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/WriteUtils.scala

```diff
@@ -54,10 +54,101 @@ import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.fs.Path
 import org.scalatest.funsuite.AnyFunSuite
 
+trait TransactionBuilderSupport {
```

Comment:
Love this!

Only note is: (a) `WriteUtils` should be moved into the `utils` package/folder, and (b) the `TransactionBuilderSupport` should be moved to a new file in that same package/folder, too.